### DISCUSSION TITLE
Is It Possible To Upgrade From Keycloak Version 21 to 22 With No Downtime on Kubernetes?

### DISCUSSION THREAD

**Discussion by xzxpurple2017** [keycloak/keycloak#26243] [Q&A]
I have a Keycloak deployment with 3 pods in the replicaset. It is currently on version 21. Is it possible to do a rolling upgrade to version 22 without any downtime? 
[keycloak-keycloak-dev-2-1705015340188944700.log](https://github.com/keycloak/keycloak/files/13954222/keycloak-keycloak-dev-2-1705015340188944700.log)

I keep running into these errors.

**Comment by vmuzikar** [comment]
Keycloak doesn't support zero-downtime upgrades unfortunately. A DB migration is performed during the upgrade which potentially contains a breaking changes to the tables.
